src/main/java/com/minecolonies/core/entity/ai/minimal/EntityAISickTask.java

```diff
@@ -410,26 +397,25 @@ private IState goToHospital()
      */
     private IState searchHospital()
     {
+        final Disease disease = citizen.getCitizenDiseaseHandler().getDisease();
```

Comment:
There was functionality to go to the hospital when low on health as well, which is why disease could be null. Did you adjust those things then?